client/my-sites/customer-home/controller.jsx

```diff
@@ -29,21 +49,21 @@ export function maybeRedirect( context, next ) {
 	}
 
 	const siteId = getSelectedSiteId( state );
-	const options = getSiteOptions( state, siteId );
+	const maybeStalelaunchpadScreenOption = getSiteOptions( state, siteId )?.launchpad_screen;
+
+	if ( maybeStalelaunchpadScreenOption && maybeStalelaunchpadScreenOption !== 'off' ) {
+		await refetchLaunchpadScreenOption( siteId, context );
+	}
```

Comment:
This condition looks incorrect. What if the site info is not yet in Redux at all? Then `getSiteOptions` will return `null` and the `maybeStalelaunchpadScreenOption` variable is `undefined`. In this case we should fetch, but we don't. This is better:
```js
if ( getSiteOption( state, siteId, 'launchpad_screen' ) !== 'off' ) {
  await context.store.dispatch( requestSite( siteId ) );
}
```
Here I also inlined the `refetchLaunchpadScreenOption` function because it's a one-liner.

Reply:
> This condition looks incorrect. What if the site info is not yet in Redux at all?

Currently the sites using launchpad are created with the screen option set, so the first time site data is loaded this should be set. I don't think we have encountered in testing any way of getting to this check without this data first being populated in some way, but maybe we are overlooking something?

I think the reason we are checking `launchpadScreenOption` value itself to be truthy is to limit this from causing every pre-existing site from always making this extra refetch/request. Every site created before or without launchpad functionality will have `launchpadScreenOption` unset and falsy, so would see `!== 'off'` will always be true for them and those sites would always be making this extra request.

Reply:
When the return value of the selector
```js
getSiteOption( state, siteId, 'launchpad_screen' )
```
is falsy, there are actually two distinct falsy values with different meanings:
1. `null` means "unknown", the site options are not yet fetched. Need to wait if you want to know the real value
2. `false` is the actual value of the option for old sites

I was considering only the first, `null` option when writing my comment. For the second option, the condition is right: we don't want to refetch when `launchpad_screen === false`.

> I don't think we have encountered in testing any way of getting to this check without this data first being populated in some way, but maybe we are overlooking something?

You're right, currently the data is guaranteed to be already populated, because the [`maybeRedirect` handler always runs after `siteSelection`](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/customer-home/index.js#L9) which has fetched the current site info.

But it's good to be aware that `null` is also a reasonable return value with a special meaning. It cannot happen today, but code can be reshuffled, refactored and moved around, and one day suddenly you have a subtle mysterious bug 🙂

Reply:
>null means "unknown", the site options are not yet fetched. Need to wait if you want to know the real value
false is the actual value of the option for old sites

Oh thanks for the clarification and distinction there, very helpful!

Reply:
[5a692ed](https://github.com/Automattic/wp-calypso/pull/67967/commits/5a692ed84df76ceb61a28624a7d0ccabadbc44a1)